ultralytics/data/annotator.py

```diff
@@ -64,3 +71,337 @@ def auto_annotate(
                     if s.any():
                         segment = map(str, s.reshape(-1).tolist())
                         f.write(f"{class_ids[i]} " + " ".join(segment) + "\n")
+
+
+class AutoAnnotator:
+    """
+    A class for automatic annotation of images using YOLO or Florence2 models.
+
+    This class handles model selection, object detection, and label generation in YOLO format.
+    It supports both YOLO-based and Florence2-based annotation pipelines, including visual output.
+
+    Attributes:
+        Y0LO_MODEL (bool): Flag indicating if the annotation uses a YOLO model.
+        model (YOLO | Florence2): Loaded model instance for annotation.
+        processor (Florence2Processor): Used only for Florence2-based annotation.
+        names (dict): Class names for YOLO-based models.
+
+    Methods:
+        annotate: Performs annotation on a dataset of images or videos.
+        convert_to_yolo: Converts bounding box coordinates to YOLO normalized format.
+        create_classes_txt: Saves class names into a `classes.txt` file.
+    """
+
+    def __init__(self, model=None, device="", variant="base"):
+        """
+        Initializes the AutoAnnotator with either a YOLO or Florence2 model.
+
+        Args:
+            model (str, optional): Model name or path. Defaults to "yolo11n.pt".
+            device (str): Device to use for inference (e.g., 'cpu', 'cuda:0').
+            variant (str): Florence2 model variant if using Florence2 (e.g., "base", "large-ft").
+        """
+        self.YOLO_MODEL = True  # Assume YOLO by default
```

Comment:
@RizwanMunawar two problems here:

- If this class supported multiple model types, then why is this attribute called YOLO_MODEL?
- Attributes should never be all uppercase. This is reserved for constants.